### PR TITLE
Standardize around typography of IDL true/false/null

### LINKED ISSUE #437: Typography: IDL true and false
In the spec, the two values of the IDL `boolean` type are currently written with `<code class=idl>` like "`true`". Yet the ECMAScript Boolean type values are written with `<emu-val>` like "**true**" in the same spec, and most other specs (e.g. the [URL Standard](https://url.spec.whatwg.org/#dom-urlsearchparams-has)) write boolean values without any particular styling like "true".

IMO we should remove the `<code class=idl>` styling. The ES vs. IDL distinction is okay (maybe even preferred).

### PR DESCRIPTION
**Pull request by TimothyGu**
I kept certain constants bolded in the spec for readability. The distinction is basically summarized in the updated Document Conventions:

> * IDL types: `unsigned long`
> * IDL values: true, false, null, 0.42, −0
> * ECMAScript classes: `Map`
> * ECMAScript language types: Object
> * ECMAScript values: generally the same notation as IDL values; bolded type (e.g. **true**) may be used when it is needed to distinguish between ECMAScript and IDL values, or if the ECMAScript value does not have a single corresponding IDL value of the same name (**undefined** and **NaN**).

The last bullet point is especially needed for things like:

> The IDL `boolean` value true is converted to the ECMAScript **true** value and the IDL boolean value false is converted to the ECMAScript **false** value.

which IMO looks odd with either both bolded or both regular-weighted. Also, ECMAScript true is usually not bolded for things like

> The property has the attributes { [[Writable]]: true, [[Enumerable]]: false, [[Configurable]]: true }.

.

Happy to have a discussion about it.

Fixes #437.

***
[Preview](https://s3.amazonaws.com/pr-preview/TimothyGu/webidl/true.html) | [Diff](https://s3.amazonaws.com/pr-preview/heycam/webidl/90efe5b...TimothyGu:b1dfa43.html)

### CODE REVIEW

**Comment by tobie** [comment]
Thanks @annevk.

Will review one last time and merge this evening (or tomorrow morning).

**Comment by annevk** [comment]
FWIW, I think the notation in OP makes sense. Hopefully at some point we can merge some Infra and IDL values, but making them look the same is already a big plus.

File: index.bs [hunk 1]
```diff
@@ -13519,9 +13522,14 @@ The following typographic conventions are used in this document:
 *   Grammar terminals: <emu-t>sometoken</emu-t>
 *   Grammar non-terminals: <emu-nt>ExampleGrammarNonTerminal</emu-t>
 *   Grammar symbols: <emu-t class="regex">identifier</emu-t>
-*   IDL types: {{unsigned long}}
-*   ECMAScript classes: {{ECMAScript/Map}}
+*   IDL types: <code class="idl">unsigned long</code> <!-- {{unsigned long}} -->
+*   IDL values: true, false, null, 0.42, −0
+*   ECMAScript classes: <code class="idl">Map</code> <!-- {{ECMAScript/Map}} -->
 *   ECMAScript language types: Object
+*   ECMAScript values: generally the same notation as IDL values; bolded type (e.g.
```
Comment (tobie) [inline index.bs]: I'm not sure we have the same perspective, here. I feel like ES values should be clearly marked as such across the spec, as they are in the ES spec.

File: index.bs [hunk 2]
```diff
@@ -11434,9 +11437,9 @@ then there must exist a property with the following characteristics:
     then the property exists on every object that implements the interface.
     Otherwise, the property exists on the [=interface prototype object=].
 *   The property has attributes
-    { \[[Writable]]: |B|, \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: |B| },
-    where |B| is <emu-val>false</emu-val> if the stringifier is [=unforgeable=] on the interface,
-    and <emu-val>true</emu-val> otherwise.
+    { \[[Writable]]: |B|, \[[Enumerable]]: true, \[[Configurable]]: |B| },
```
Comment (tobie) [inline index.bs]: I'd rather we stayed consistent with the ES spec, which marks-up ES values within property descriptors.

**Comment by tobie** [comment]
Sorry I rebased this. I had planned to just merge it before I noticed this was broader than just fixing incorrectly marked-up IDL values. Hope it's not too much of a hindrance.

**Comment by TimothyGu** [comment]
@tobie The main consideration in not marking up "ECMAScript booleans" as different from "IDL booleans" is https://github.com/whatwg/infra/issues/95. I expect Infra to eventually define booleans in a similar way to Infra lists (which is explicitly stated to be equivalent to ECMAScript Lists):

> Whenever JavaScript expects a List, a list as defined here can be used; they are the same type.

If that happens, IDL booleans will be the same things as ECMAScript booleans (which makes sense), both of which are going to be mapped directly to Infra booleans.

Also, the ECMAScript-based pieces in HTML don't capitalize "true", "false", "null", or even "undefined" at all. See e.g. [The `WindowProxy` exotic object](https://html.spec.whatwg.org/multipage/window-object.html#the-windowproxy-exotic-object).

**Comment by tobie** [comment]
> The main consideration in not marking up "ECMAScript booleans" as different from "IDL booleans" is whatwg/infra#95. I expect Infra to eventually define booleans in a similar way to Infra lists (which is explicitly stated to be equivalent to ECMAScript Lists):

Can we wait for those until that's clarified in infra?

> Also, the ECMAScript-based pieces in HTML don't capitalize "true", "false", "null", or even "undefined" at all. See e.g. The WindowProxy exotic object.

s/capitalize/markup/ ? If so, I see what you mean. But it still seems bizarre to create our own flavor of ES markup, while style relying on the same tags the ES spec is using.

**Comment by annevk** [comment]
@TimothyGu I think you're basically correct and I'd be happy to merge this. But at this point it needs rebasing. Is this still something you want to pursue?

**Pull request closed without merging**